### PR TITLE
Getting Started with APM

### PR DESCRIPTION
**Pull request by brett0000FF**
### What does this PR do? What is the motivation?

- Getting Started with Tracing hasn't been updated in a few years.
- It currently includes outdated commands and doesn't mention Single Step Instrumentation.
- This update refines the guide to match current positioning, uses the latest commands, and uses Single Step Instrumentation.
- Additionally, this guide presents the flow of basic -> advanced setup, starting with Single Step, then offering custom instrumentation as an advanced option.

TODO
- [x] Mention that single step is in beta if we publish this before it GAs.
- [x] Update menu.
- [x] Rename to Getting Started with APM Tracing?
- [x] Highlight correct lines in custom instrumentation example.
- [x] Amend custom instrumentation step to include installing ddtrace via plain-old-language-specific-package-manager.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/getting_started/tracing/_index.md
Comment (domalessi) [inline content/en/getting_started/tracing/_index.md:89]: I believe these tick marks shouldn't be there? Typo?

File: content/en/getting_started/tracing/_index.md
Comment (domalessi) [inline content/en/getting_started/tracing/_index.md:48]:
```suggestion
    DD_API_KEY=<YOUR_DD_API_KEY> DD_SITE="<YOUR_DD_SITE>" DD_APM_INSTRUMENTATION_ENABLED=host DD_ENV=<AGENT_ENV> bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script_agent7.sh)"
```

File: content/en/getting_started/tracing/_index.md
Comment (brett0000FF) [reply content/en/getting_started/tracing/_index.md]: Good catch!

File: content/en/getting_started/tracing/_index.md
Comment (vidurkhannadatadog) [inline content/en/getting_started/tracing/_index.md:27]: you will*

File: content/en/getting_started/tracing/_index.md
Comment (vidurkhannadatadog) [inline content/en/getting_started/tracing/_index.md:47]: I don't think I follow the flow. Setting up an application should be done before running single step.

File: content/en/getting_started/tracing/_index.md
Comment (vidurkhannadatadog) [inline content/en/getting_started/tracing/_index.md:141]: Let' call it traces. Observability data is very generic

and let' attach a screen shot of traces too.

File: content/en/getting_started/tracing/_index.md
Comment (jonbodner) [inline content/en/getting_started/tracing/_index.md:43]: Maybe "without needing to modify your application's code or the deployment process"? Want to emphasize that nothing is changed other than installing the datadog agent.

File: content/en/getting_started/tracing/_index.md
Comment (jonbodner) [inline content/en/getting_started/tracing/_index.md:70]: "Add the highlighted lines to the code in `hello.py`" ?

**Review by jonbodner — APPROVED** [review]
minor suggestions, can make these changes or not without additional approval.

At some point, I want to remove setting DD_SERVICE from the example, but today is not that day.

File: content/en/getting_started/tracing/_index.md
Comment (avivenzio-dd) [inline content/en/getting_started/tracing/_index.md:76]: wondering if it makes sense to only install the `python` library as a part of this guide as thats what the example code is written in? It might speed up the install and less potential to break things. Unless part of the goal here is to expose users to things that we found that they did not explicitly intend to instrument?

@vidurkhannadatadog @jonbodner any thoughts?

**Review by avivenzio-dd — COMMENTED** [review]
one non-blocking thought/question. but looks great!

File: content/en/getting_started/tracing/_index.md
Comment (domalessi) [inline content/en/getting_started/tracing/_index.md:158]:
```suggestion
To illustrate this, you will import the Datadog Python tracing library into `hello.py` and create a custom span and span tag.
```

File: content/en/getting_started/tracing/_index.md
Comment (domalessi) [inline content/en/getting_started/tracing/_index.md:168]:
```suggestion
1. Add the highlighted lines to the code in `hello.py` to create a custom span tag `get_quote` and a custom span tag `quote`:
```

Or something like that, just to be a little more explicit about what we're doing here.

File: content/en/getting_started/tracing/_index.md
Comment (domalessi) [inline content/en/getting_started/tracing/_index.md:154]: I don't love the "This is great" phrasing -- feels a bit vague -- but not sure how to make it much better lol. I want to do something like "This is the preferred approach if...". Dunno if that's better at all. Defer to you!

File: content/en/getting_started/tracing/_index.md
Comment (domalessi) [inline content/en/getting_started/tracing/_index.md:204]:
```suggestion
1. Find the new custom `get_quote` span in the flame graph and hover over it:
```

File: content/en/getting_started/tracing/_index.md
Comment (brett0000FF) [reply content/en/getting_started/tracing/_index.md:79]: Good catch! These are not pointing where I intended them to.

File: content/en/getting_started/tracing/_index.md
Comment (brett0000FF) [reply content/en/getting_started/tracing/_index.md:154]: Good thinking. I will rephrase this slightly.

File: content/en/getting_started/tracing/_index.md
Comment (brett0000FF) [reply content/en/getting_started/tracing/_index.md:109]: I see where you are coming from, but I'd like to leave these details off as they are not connected to the main goal of the guide.

**Pull request merged**
